### ISSUE TITLE
How to define custom "actions"?

### ISSUE THREAD

**Issue by zikalino** [Azure/magic-module-specs#21]
Should api.yaml allow defining custom actions? They could be for example "start", "stop", "generalize", etc.
They have different URL in REST API, for instance "…../virtualMachines/myVm/stop", and they have separate method in SDK as well.
We should have schema for that so autorest.devops can generate api.yaml correctly.

**Comment by JunyiYi** [comment]
My design would be:

1. do not define any custom operations here (user provides their customized API call in Python code in ansible.yaml).
2. But in api.yaml, we need to define the `status` property of string enum containing `Started`, `Stopped`. User would also provide their customized code snippet in ansible.yaml to link the enum value with the actual api call.

**Comment by zikalino** [comment]
If we do (1) it won't be automatic code generation, we will have to include the same custom code again and again for different azure resources.
I think we should have proper operations defined, then we should have default handling for these, and if it doesn't work as we want we should create custom code.

**Comment by JunyiYi** [comment]
@zikalino Yes you are right. I do not plan to make it automatic for the following reasons:

1. Terraform does not have this requirements.
2. Not many resources needs this behavior.

So why not for now let's using manual customization code snippet. If in the future we found there are so many similar patterns, we can generate those customization code snippets, instead of putting everything into api.yaml since terraform does not need it.

**Comment by zikalino** [comment]
@JunyiYi 
So we should resolve some basics, here are several unknowns:
- If I add properties for "started" / "stopped", etc… to the UX spec, should they have no references to the SDK? If they won't have valid references, won't it lead to MM error or something like that?
- I think we should add methods, like "start", etc. co the custom code you create has some connection point
I think we should agree on some basic framework for defining these things.

I think this feature can be postponed a bit, maybe hurrying to do it now will result in something we will have to undo in the future.

@yungezz what do you think?

**Comment by JunyiYi** [comment]
@zikalino I agree we should carefully design the feature. My two cents are, if a feature is only supported in one specific product (no matter Terraform or Ansible), then it should not be put in api.yaml.

For now, maybe customization is the standard work-around to unblock the scenario? I think you can just make `azure_sdk_references: []` an empty array, I will fix magic-modules to support that.
